### PR TITLE
Fix 474, support bytes in both hex and plain strings

### PR DESCRIPTION
**Pull request by hewigovens**

### CODE REVIEW

File: Trust/EtherClient/Types/EthTypedData.swift [hunk 1]
```diff
@@ -86,7 +86,9 @@ struct EthTypedData: Decodable {
             return getTypedData(for: int, align: size)
         case .string(let string):
             if type.starts(with: "bytes") {
-                return Data(hex: string)
+                if string.isHexEncoded {
+                    return Data(hex: string)
+                }
```
Comment (vikmeup) [inline Trust/EtherClient/Types/EthTypedData.swift]: what happens if it's `bytes` and not `isHexEncoded `?

File: Trust/Extensions/String.swift [hunk 2]
```diff
@@ -13,6 +13,17 @@ extension String {
         return data.hexEncoded
     }
 
+    var isHexEncoded: Bool {
```
Comment (vikmeup) [inline Trust/Extensions/String.swift]: Wonder if there is better way to handle this

File: Trust/EtherClient/Types/EthTypedData.swift (hunk 1 above)
Comment (hewigovens) [reply Trust/EtherClient/Types/EthTypedData.swift]: fallback to plain string, we have some tests covered.

File: Trust/Extensions/String.swift (hunk 2 above)
Comment (hewigovens) [reply Trust/Extensions/String.swift]: something similar to: https://github.com/ethjs/ethjs-util/blob/a76d4798e18ab78aabd1bb91cef2ebe0b9f2863c/src/index.js#L192

**Pull request merged**